railgun/internal/diagnostics/server.go

```diff
@@ -46,8 +63,28 @@ func (s *Server) Listen(ctx context.Context) error {
 	}
 }
 
-// installHandlers adds the Profiling webservice to the given mux.
-func installHandlers(mux *http.ServeMux) {
+// receiveConfig watches the config update channel
+func (s *Server) receiveConfig(ctx context.Context) {
+	for {
+		select {
+		case dump := <-s.ConfigDumps.Configs:
+			if !dump.Failed {
+				successfulConfigDump = dump.Config
+			} else {
+				failedConfigDump = dump.Config
+			}
```

Comment:
nit: I'm personally more in favor of not using negative assertions as the leader in branch logic, as I feel this reads clearer at a glance:
```suggestion
			if dump.Failed {
				failedConfigDump = dump.Config
			} else {
				successfulConfigDump = dump.Config
			}
```